Mathlib/NumberTheory/ModularForms/LevelOne.lean

```diff
@@ -57,28 +57,33 @@ namespace ModularFormClass
 
 variable [ModularFormClass F Γ(1) k]
 
+lemma one_mem_strictPeriods_SL2Z : (1 : ℝ) ∈ Γ(1).strictPeriods := by
+  rw [Gamma_one_top, ← MonoidHom.range_eq_map, Subgroup.strictPeriods_SL2Z]
+  exact AddSubgroup.mem_zmultiples 1
+
 private theorem cuspFunction_eqOn_const_of_nonpos_wt (hk : k ≤ 0) (f : F) :
     Set.EqOn (cuspFunction 1 f) (const ℂ (cuspFunction 1 f 0)) (Metric.ball 0 1) := by
   refine eq_const_of_exists_le (fun q hq ↦ ?_) (exp_nonneg (-π)) ?_ (fun q hq ↦ ?_)
-  · exact (differentiableAt_cuspFunction 1 f (mem_ball_zero_iff.mp hq)).differentiableWithinAt
+  · exact (differentiableAt_cuspFunction f one_pos one_mem_strictPeriods_SL2Z
+      (mem_ball_zero_iff.mp hq)).differentiableWithinAt
   · simp only [exp_lt_one_iff, Left.neg_neg_iff, pi_pos]
   · simp only [Metric.mem_closedBall, dist_zero_right]
     rcases eq_or_ne q 0 with rfl | hq'
     · refine ⟨0, by simpa only [norm_zero] using exp_nonneg _, le_rfl⟩
     · obtain ⟨ξ, hξ, hξ₂⟩ := exists_one_half_le_im_and_norm_le hk f
         ⟨_, im_invQParam_pos_of_norm_lt_one Real.zero_lt_one (mem_ball_zero_iff.mp hq) hq'⟩
       exact ⟨_, norm_qParam_le_of_one_half_le_im hξ,
-        by simpa only [← eq_cuspFunction 1 f, Nat.cast_one, coe_mk_subtype,
-          qParam_right_inv one_ne_zero hq'] using hξ₂⟩
+        by simpa only [← eq_cuspFunction f _ one_mem_strictPeriods_SL2Z one_ne_zero,
+          Nat.cast_one, coe_mk_subtype, qParam_right_inv one_ne_zero hq'] using hξ₂⟩
 
 private theorem levelOne_nonpos_wt_const (hk : k ≤ 0) (f : F) :
-    ⇑f = Function.const _ (cuspFunction 1 f 0) := by
+    f = Function.const ℍ (cuspFunction 1 f 0) := by
   ext z
   have hQ : 𝕢 1 z ∈ (Metric.ball 0 1) := by
     simpa only [Metric.mem_ball, dist_zero_right, neg_mul, mul_zero, div_one, Real.exp_zero]
       using (norm_qParam_lt_iff zero_lt_one 0 z.1).mpr z.2
-  simpa only [← eq_cuspFunction 1 f z, Nat.cast_one, Function.const_apply] using
-    (cuspFunction_eqOn_const_of_nonpos_wt hk f) hQ
+  simpa only [← eq_cuspFunction f _ one_mem_strictPeriods_SL2Z one_ne_zero, Nat.cast_one,
+    Function.const_apply] using cuspFunction_eqOn_const_of_nonpos_wt hk f hQ
```

Comment:
One could replace this terminal `simpa only` (and others as well) by `simpa`, e.g., here

```suggestion
  simpa [← eq_cuspFunction f _ one_mem_strictPeriods_SL2Z one_ne_zero]
    using cuspFunction_eqOn_const_of_nonpos_wt hk f hQ
```
(which is, I think, nowadays preferred).